example/main.mo

```diff
@@ -0,0 +1,255 @@
+import Array "mo:base/Array";
+import Error "mo:base/Error";
+import Int "mo:base/Int";
+import Prim "mo:prim";
+import Principal "mo:base/Principal";
+import Text "mo:base/Text";
+import Timer "mo:base/Timer";
+import Vec "mo:vector";
+
+import ICRC1 "../src/ICRC1";
+import TokenHandler "../src";
+
+actor class Example() = self {
+
+  stable var assetsData : Vec.Vector<StableAssetInfo> = Vec.new();
+
+  type AssetInfo = {
+    ledgerPrincipal : Principal;
+    handler : TokenHandler.TokenHandler;
+  };
+  type StableAssetInfo = {
+    ledgerPrincipal : Principal;
+    handler : TokenHandler.StableData;
+  };
+
+  type TokenInfo = {
+    min_deposit : Nat;
+    min_withdrawal : Nat;
+    deposit_fee : Nat;
+    withdrawal_fee : Nat;
+  };
+
+  type NotifyResult = {
+    #Ok : {
+      deposit_inc : Nat;
+      credit_inc : Nat;
+    };
+    #Err : {
+      #CallLedgerError : Text;
+      #NotAvailable : Text;
+    };
+  };
+
+  type WithdrawResult = {
+    #Ok : {
+      txid : Nat;
+      amount : Nat;
+    };
+    #Err : {
+      #CallLedgerError : Text;
+      #InsufficientCredit;
+      #AmountBelowMinimum;
+    };
+  };
```

Comment:
We should export these type from TokenHandler so that the example code does not have to define them.

For consistent naming better to call these NotifyRespose instead of NotifyResult.

Reply:
You are right. The types are related to icrcX, not to TokenHandler. So we could make a file icrcX.mo in the future to import these files.